oidc_auth/settings.py

```diff
@@ -1,13 +1,17 @@
 from django.conf import settings
 from rest_framework.settings import APISettings
 
-
 USER_SETTINGS = getattr(settings, 'OIDC_AUTH', None)
 
 DEFAULTS = {
     'OIDC_ENDPOINT': None,
-    'OIDC_ENDPOINTS': {},
-    'OIDC_AUDIENCES': None,
+
+    # Currently unimplemented
+    'OIDC_ENDPOINTS': [],
+
+    # The Claims Options can now be defined by a static string.
+    # ref: https://docs.authlib.org/en/latest/jose/jwt.html#jwt-payload-claims-validation
+    'OIDC_CLAIMS_OPTIONS': {},
```

Comment:
Hmm, interestingly this does not seem to break for me if I do not change it in my local running version thinking I.e. the change from OIDC_AUDIENCES to OIDC_CLAIMS_OPTIONS, I would have expected I needed to make some changes locally. But whatever I do does not seem to have an effect to the outcome (i.e. it just works smile ). Did you encouter any errors?

Reply:
Could you specify of what you mean with "whatever you change".
I currently only have some errors with integrating it into my application.

@kerrermanisNL I think for testing the handling of `azp` as an example we need to have different settings for the `OIDC_CLAIMS_OPTIONS`.
But from the test perspective I think most of the stuff just works.

Reply:
We could add backwards compatibility if only `OIDC_AUDIENCES` is defined to build it into the `OIDC_CLAIMS_OPTIONS`, so that older configs work as expected.

Reply:
For me the integration into the application works fine, but that's if I don't change `OIDC_AUDIENCES` (which should no longer exist). I would expect the API not to handle any audiences since the `OIDC_AUDIENCES` is not being used to check which audiences is valid. But for some reason it still accepts the audiences defined in there. So I'm wondering why that is. Either I'm doing something wrong or something is not right. I did check the version I was running of drf-oidc-auth and it was the correct one. So what I'm trying to say is that I'm surprised my application works with no changes to the settings of drf-oidc-auth :thinking:

Reply:
I think something changed in regards to the `OIDC_CLAIMS` behaviour. Previously if an audience was not in the `OIDC_AUDIENCES` an error would be returned (see https://github.com/ByteInternet/drf-oidc-auth/blob/de85585d3a4f742aedf77ef8fd8d187cf893bd51/oidc_auth/authentication.py#L153). Right now I think any audience is able to make a request independent of which audiences are specified in the settings. Because I have not specified any audiences in the settings but am still able to make a request and receive a response independent on which audience does the request.

Reply:
@kerrermanisNL I will check for the audience bug :) I will create a test for it so we could check when not defined what happens and should happen.

Reply:
Cool, let me know how it goes! :)

Reply:
I just double checked and it indeed seems that if no audiences are provided all audiences are allowed. I think that differs from the previous implementation. However, if you _do_ provide some audiences it works correctly. So there should be something that returns "invalid audience" if there are no audiences :)

Reply:
@kerrermanisNL ok i will add that functionality ;)

Reply:
Sweet! :dog2:

Reply:
@kerrermanisNL I checked about the audience bug and what you need to do is set the claim `aud` as `essential: True` then the claim is checked on missing.

Reply:
Ok, cool! It seems like this is different behaviour to the previous versions though. It would probably be nice if we were able to keep the same functionality. It might be unexpected behaviour, even if people upgrade a major version. I would not be surprised if people see no change in behaviour they think everything is working accordingly, even though we now silently ignore audiences. 

Should probably mention this in the docs as required as well now.